test/AbandonTest.t.sol

```diff
@@ -0,0 +1,13 @@
+pragma solidity ^0.8.13;
+
+import "./AbandonAbstract.t.sol";
+import "forge-std/Test.sol";
+
+contract AbandonTest is AbandonAbstract {
+
+    function testSonicWasNice() public {
+        vm.prank(bubsy);
+        r.registerName("Bubsy");
```

Comment:
You should have test coverage for all of your events. You don't need to test every event that gets emitted in all of your tests, just one per event type.  

Foundry vm.expectEmit is a little tricky.  Spend a little time on it but if you can't get it working, please reach out on slack.